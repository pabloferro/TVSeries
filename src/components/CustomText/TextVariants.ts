export enum TextVariants {
  default = 'default',
  h1 = 'h1',
  h2 = 'h2',
}
