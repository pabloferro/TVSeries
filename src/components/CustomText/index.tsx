import React from 'react';
// eslint-disable-next-line no-restricted-imports
import {Text, TextProps} from 'react-native';

import variants from './variantsStyles';
import {TextVariants} from './TextVariants';

interface Props extends TextProps {
  variant?: TextVariants;
}

export default function CustomText({variant, style, ...props}: Props) {
  return <Text style={[variants[variant || 'default'], style]} {...props} />;
}
