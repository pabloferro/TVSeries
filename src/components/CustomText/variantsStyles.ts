import {StyleSheet, TextStyle} from 'react-native';
import {secondary} from '../../constants/colors';
import {TextVariants} from './TextVariants';

const baseStyle: TextStyle = {
  color: 'white',
};

export default StyleSheet.create<{
  [key in TextVariants]: TextStyle;
}>({
  default: {
    ...baseStyle,
  },
  h1: {
    ...baseStyle,
    fontSize: 24,
    fontWeight: '700',
  },
  h2: {
    ...baseStyle,
    fontSize: 18,
    fontWeight: '500',
  },
  body: {
    ...baseStyle,
    fontSize: 16,
    fontWeight: '400',
  },
  secondary: {
    ...baseStyle,
    color: secondary,
    fontSize: 16,
    fontWeight: '500',
  },
});
