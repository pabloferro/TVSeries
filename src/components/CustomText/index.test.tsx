import React from 'react';
import {render} from '@testing-library/react-native';

import {TextVariants} from './TextVariants';
import CustomText from '.';

test.each(Object.values(TextVariants))(
  'variant %s renders correctly',
  variant => {
    const {toJSON} = render(<CustomText variant={variant}>Text</CustomText>);
    expect(toJSON()).toMatchSnapshot();
  },
);

test('without variant renders correctly', () => {
  const {toJSON} = render(<CustomText>Text</CustomText>);
  expect(toJSON()).toMatchSnapshot();
});
