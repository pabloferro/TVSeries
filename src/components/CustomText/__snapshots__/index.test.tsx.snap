// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`variant body renders correctly 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#FFF",
        "fontSize": 16,
        "fontWeight": "400",
      },
      undefined,
    ]
  }
>
  Text
</Text>
`;

exports[`variant default renders correctly 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#FFF",
      },
      undefined,
    ]
  }
>
  Text
</Text>
`;

exports[`variant h1 renders correctly 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#FFF",
        "fontSize": 24,
        "fontWeight": "700",
      },
      undefined,
    ]
  }
>
  Text
</Text>
`;

exports[`variant h2 renders correctly 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#FFF",
        "fontSize": 18,
        "fontWeight": "500",
      },
      undefined,
    ]
  }
>
  Text
</Text>
`;

exports[`variant secondary renders correctly 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#999",
        "fontSize": 16,
        "fontWeight": "500",
      },
      undefined,
    ]
  }
>
  Text
</Text>
`;

exports[`without variant renders correctly 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#FFF",
      },
      undefined,
    ]
  }
>
  Text
</Text>
`;
